gems/smithy/lib/smithy/templates/client/protocol_plugin.erb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module <%= module_name %>
+  module Plugins
+     # Protocol plugin
+     class Protocol < Smithy::Client::Plugin
+       option(
+         :protocol,
+         doc_default: 'Smithy::Client::Protocols::<%= protocol_name %>',
+         doc_type: 'Smithy::Client::Protocols::ProtocolBase',
+         docstring: 'TODO'
+         ) do |_cfg|
+           Smithy::Client::Protocols::<%= protocol_name %>.new
+         end
+
+       # @api private
+       class Build < Smithy::Client::Handler
```

Comment:
I know these are super simple, but I think I'd lean towards having the handlers be in smithy-client and minimizing the amount of codegen.  (and we should maybe do that with endpoint plugin as well?)

Reply:
Yeah - after reading through the discussion points - agreed, these handlers are not necessary to be codegenerated. I will rework here.

Reply:
Since these handlers are very generic - I was thinking about moving them into its own plugin class within `smithy-client` level but... what do I call it? `BuildParsePlugin`? Naming is hard but ideally I want a single plugin that adds these generic handlers. But I was also thinking of doing separate plugins like `Build` or `Parse` - kind of like `Sign` plugin that we already have. 

 Alternatively - I could leave it here for starters.

Reply:
As an alternative - you could just define the handler classes in smithy-client, but have the generated plugin add them.